### ISSUE TITLE
Serious performance degradation in Google Docs when disabled

### ISSUE THREAD

**Issue by nkgm** [philc/vimium#2134]
To reproduce, try opening a decent sized google doc with a few pictures in it and press the down arrow key. Now try with the extension globally disabled. As long as the extension is enabled, google docs will be sluggish, and adding an exception rule (disabling vimium) for the page doesn't make any difference.

**Comment by smblott-github** [comment]
Have you got an example document, @nkgm?

**Comment by nkgm** [comment]
Try [this google doc](https://drive.google.com/open?id=1oIGrIH6wbll3AcCOW8tYp-BrWRJX2k35UnIwUig30ZE) for example. On my MBP with all extensions disabled it's not entirely smooth, but still usable. With vimium extension enabled (but still disabled for the page) it lags so bad it's unusable for me.

**Comment by smblott-github** [comment]
Thanks, @nkgm.  I don't see any sluggishness.  What's "not smooth" for you?

**Comment by nkgm** [comment]
> To reproduce, try opening a decent sized google doc with a few pictures in it and press the down arrow key.

So basically disable all extensions but vimium (even add a vimium exception for docs.google.com/* so it's active but disabled for the page), place the caret at the start of the document and keep the down arrow key pressed (40ms repeat rate here) and observe the sluggishness. Now try the same with the extension disabled in the chrome extensions page. Can do screencast if needed.

**Comment by smblott-github** [comment]
Thanks, @nkgm.  I *can* reproduce that now.  It's not immediately obvious what's going on.  Even disabling (that is, not installing) Vimium's main event handlers doesn't make the effect go away.

Note, however, the Google Docs scrolling operation does seem to be pretty processor intensive.  On my laptop, the fan spins up after just a few seconds.  So, *any* extra work Vimium's doing would be disruptive.

**Comment by nkgm** [comment]
Thanks for investigating @smblott-github. Indeed the scrolling operation is very intensive and I'm not suggesting vimium should do something that's clearly not possible. I was only hoping there was a way to cleanly override the plugin in this case.

Could it just be the overhead of reinitializing the plugin every time the url fragment changes? Is it possible to profile something like this?

**Comment by gdh1995** [comment]
I also reproduced the problem.

If I use Developer Tools to show the fps number:
* Normally, the page has a frame rate of 18 fps, which means only the cursor is blinking.
* If I've turned off Vimium, and begin to scroll up from the bottom, the frame rate will be between 20 ~ 45 fps (desktop computer, Intel i5-4590).
* When Vimium is enabled without any exclusion rule, the frame rate is about 10 fps firstly, and reduces rapidly to 0.9fps ~ 3fps.
* After disabling VImium on chrome://extensions (not refreshing the doc page), its frame rate is 30~40 fps, and reduces to about 25~30 fps.

It seems that hash change events will cause too many messages, which can not be handled in time, and many things including redrawing are delayed heavily.
